code/game/machinery/medical_kiosk.dm

```diff
@@ -0,0 +1,345 @@
+	..()
+	default_unfasten_wrench(user, I, time = 10)
+	return TRUE
+
+/obj/machinery/medical_kiosk/RefreshParts()
+	var/obj/item/circuitboard/machine/medical_kiosk/board = circuit
+	if(board)
+		active_price = board.custom_cost
+	return
+
+/obj/machinery/medical_kiosk/attackby(obj/item/O, mob/user, params)
+	if(default_deconstruction_screwdriver(user, "kiosk_open", "kiosk", O))
+		return
+	else if(default_deconstruction_crowbar(O))
+		return
+
+	if(istype(O, /obj/item/scanner_wand))
+		var/obj/item/scanner_wand/W = O
+		if(scanner_wand)
+			to_chat(user, "<span class='warning'>There's already a scanner wand in [src]!</span>")
+			return
+		if(HAS_TRAIT(O, TRAIT_NODROP) || !user.transferItemToLoc(O, src))
+			to_chat(user, "<span class='warning'>[O] is stuck to your hand!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] snaps [O] onto [src]!</span>", \
+		"<span class='notice'>You press [O] into the side of [src], clicking into place.</span>")
+		 //This will be the scanner returning scanner_wand's selected_target variable and assigning it to the altPatient var
+		if(W.selected_target)
+			if(!(altPatient == W.return_patient()))
+				clearScans()
+			altPatient = W.return_patient()
+			user.visible_message("<span class='notice'>[W.return_patient()] has been set as the current patient.</span>")
+			W.selected_target = null
+		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+		scanner_wand = O
+		return
+	return ..()
+
+/obj/machinery/medical_kiosk/AltClick(mob/living/carbon/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE))
+		return
+	if(!scanner_wand)
+		to_chat(user, "<span class='warning'>The scanner wand is currently removed from the machine.</span>")
+		return
+	if(!user.put_in_hands(scanner_wand))
+		to_chat(user, "<span class='warning'>The scanner wand falls to the floor.</span>")
+		scanner_wand = null
+		return
+	user.visible_message("<span class='notice'>[user] unhooks the [scanner_wand] from [src].</span>", \
+	"<span class='notice'>You detach the [scanner_wand] from [src].</span>")
+	playsound(src, 'sound/machines/click.ogg', 60, TRUE)
+	scanner_wand = null
+
+/obj/machinery/medical_kiosk/Destroy()
+	qdel(scanner_wand)
+	return ..()
+
+/obj/machinery/medical_kiosk/emag_act(mob/user)
+	..()
+	if(obj_flags & EMAGGED)
+		return
+	if(user)
+		user.visible_message("<span class='warning'>[user] waves a suspicious card by the [src]'s biometric scanner!</span>",
+	"<span class='notice'>You overload the sensory electronics, the diagnostic readouts start jittering across the screen..</span>")
+	obj_flags |= EMAGGED
+	var/obj/item/circuitboard/computer/cargo/board = circuit
+	board.obj_flags |= EMAGGED //Mirrors emag status onto the board as well.
+	pandemonium = TRUE
+
+/obj/machinery/medical_kiosk/examine(mob/user)
+	. = ..()
+	if(scanner_wand == null)
+		. += "<span class='notice'>\The [src] is missing its scanner.</span>"
+	else
```

Comment:
Why alt click to remove the scanner when this has a UI?
Perhaps make it a button on the UI, or give the machine a radial menu?